### PR TITLE
support FDB's release candidate versioning scheme

### PR DESCRIPTION
**Pull request by ltsampros**
# Description

*Please include a summary of the change and which issue is addressed. If this change resolves an issue, please include the issue number in the description.*

## Type of change

*Please select one of the options below.*

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Documentation
- Other

# Discussion

*Are there any design details that you would like to discuss further?*

# Testing

*Please describe the tests that you ran to verify your changes. Unit tests?
Manual testing?*

*Do we need to perform additional testing once this is merged, or perform in a larger testing environment?*

# Documentation

*Did you update relevant documentation within this repository?*

*If this change is adding new functionality, do we need to describe it in our user manual?*

*If this change is adding or removing subreconcilers, have we updated the core technical design doc to reflect that?*

*If this change is adding new safety checks or new potential failure modes, have we documented and how to debug potential issues?*

# Follow-up

*Are there any follow-up issues that we should pursue in the future?*

*Does this introduce new defaults that we should re-evaluate in the future?*

### CODE REVIEW

**Review by brownleej — APPROVED** [review]
👍

**Review by johscheuer — APPROVED** [review]
Just some comments from my side (nothing blocking once you fixed them feel free to merge). On another side note I wonder if we should use a library for this in the future e.g. https://github.com/Masterminds/semver or https://pkg.go.dev/golang.org/x/mod/semver (the latter would require to prefix a version with with a `v`) instead of having our own implementation.

File: api/v1beta1/foundationdb_version.go [hunk 1]
```diff
@@ -39,10 +39,12 @@ type FdbVersion struct {
 
 	// Patch is the patch version
 	Patch int
+
+	ReleaseCandidate int
 }
 
 // FDBVersionRegex describes the format of a FoundationDB version.
-var FDBVersionRegex = regexp.MustCompile(`(\d+)\.(\d+)\.(\d+)`)
+var FDBVersionRegex = regexp.MustCompile(`(\d+)\.(\d+)\.(\d+)(-rc(\d+))?`)
```
[the remark this answers is not in the document]
Comment (ltsampros) [reply api/v1beta1/foundationdb_version.go:47]: Yes, this was discussed with @ammolitor and we have agreed on positive numbers for the `-rcN` specifier.

**Comment by ltsampros** [comment]
> Just some comments from my side (nothing blocking once you fixed them feel free to merge). On another side note I wonder if we should use a library for this in the future e.g. https://github.com/Masterminds/semver or https://pkg.go.dev/golang.org/x/mod/semver (the latter would require to prefix a version with with a `v`) instead of having our own implementation.

I think we should use one of those libraries if/when semantic versioning is an accepted requirement on the FDB side.

**Pull request merged**